odoo/addons/base/models/res_bank.py

```diff
@@ -101,7 +101,7 @@ def _compute_acc_type(self):
         for bank in self:
             bank.acc_type = self.retrieve_acc_type(bank.acc_number)
 
-    @api.depends('partner_id')
+    @api.depends('partner_id.name')
```

Comment:
As the account holder name is readonly False, so it can still be adapted and manipulated by the user, I am not sure it is the best idea, to magically change it when the name of the partner changes.

Reply:
Checking deeper, as the acc_holder_name field is not even in any view in standard, most users will be annoyed in their SEPA, ... by not having it changed than there will be users that will be annoyed by a magical change on the partner when they did the force change with a field in the view.  So, we should go forward with it.  We could also change it to False, but that is a bigger change than this one as well wrt stable.